framework/math/linear_solver/linear_solver.h

```diff
@@ -78,29 +48,54 @@ class LinearSolver
   void SetKSPSolveSuppressionFlag(bool flag) { suppress_kspsolve_ = flag; }
   bool GetKSPSolveSuppressionFlag() const { return suppress_kspsolve_; }
 
+  /**
+   * Set up the linaer solver
+   */
+  virtual void Setup();
+
+  /**
+   * Solve the system
+   */
+  virtual void Solve();
+
 protected:
+  bool IsSystemSet() const { return system_set_; }
   virtual void PreSetupCallback();
   virtual void SetOptions();
   virtual void SetSolverContext();
   virtual void SetConvergenceTest();
   virtual void SetMonitor();
   virtual void SetPreconditioner();
-
   virtual void SetSystemSize() = 0;
   virtual void SetSystem() = 0;
   virtual void PostSetupCallback();
-
-public:
-  virtual void Setup();
-
-protected:
   virtual void PreSolveCallback();
   virtual void SetInitialGuess() = 0;
   virtual void SetRHS() = 0;
   virtual void PostSolveCallback();
 
-public:
-  virtual void Solve();
+  /// Solver name
+  const std::string solver_name_;
+  const std::string iterative_method_;
+
+  LinSolveContextPtr context_ptr_ = nullptr;
+
+  /// Matrix
+  Mat A_;
+  /// Right-hand side
+  Vec b_;
+  /// Solution vector
+  Vec x_;
+  /// PETSc KSP object
+  KSP ksp_;
+  /// Number of local DOFs in the system
+  int64_t num_local_dofs_ = 0;
+  /// Number of global DOFs in the system
+  int64_t num_globl_dofs_ = 0;
```

Comment:
```suggestion
  int64_t num_global_dofs_ = 0;
```

Reply:
I agree with global being fully spelled out. It's not our fault that local has one less letter than global!

I've used n_xyz for number of xyz. I think n_ is sufficient compared to num_. It's just my opinion. No action needed.